neural_lam/utils.py

```diff
@@ -263,11 +262,65 @@ def fractional_plot_bundle(fraction):
     return bundle
 
 
-def init_wandb_metrics(wandb_logger):
+def init_wandb_metrics(wandb_logger, val_steps):
     """
     Set up wandb metrics to track
     """
     experiment = wandb_logger.experiment
     experiment.define_metric("val_mean_loss", summary="min")
-    for step in constants.VAL_STEP_LOG_ERRORS:
+    for step in val_steps:
         experiment.define_metric(f"val_loss_unroll{step}", summary="min")
+
+
+class ConfigLoader:
+    """
+    Class for loading configuration files.
+
+    This class loads a YAML configuration file and provides a way to access
+    its values as attributes.
+    """
+
+    def __init__(self, config_path, values=None):
```

Comment:
you could create a classmethod called something like `Config.from_file(filepath)`, e.g.

```python
class Config
    def __init__(self, values):
        self.values = values

class YamlConfig(Config):
    @classmethod
    def from_file(cls, filepath):
        with open(self.config_path, encoding="utf-8", mode="r") as file:
            return cls(values=yaml.safe_load(file))
```

This makes it possible to create a `Config` directly from a `dict` for example and allows us to easily implement other storage formats. Or alternatively:

```python
class Config
    def __init__(self, values):
        self.values = values

    @classmethod
    def from_file(cls, filepath):
        if filepath.endswith(".yaml"):
            with open(self.config_path, encoding="utf-8", mode="r") as file:
                return cls(values=yaml.safe_load(file))
        else:
             raise NotImplementedError(Path(filepath).suffix))
```

I think I prefer the latter actually since it means fewer classes, and it probably isn't important to know exactly what format the config was loaded from.